compiler/pash_runtime.py

```diff
@@ -721,14 +722,20 @@ def add_eager_nodes(graph, use_dgsh_tee):
     intermediateFileIdGen = FileIdGen(0, runtime_config['eager_intermediate_prefix'])
 
     ## Get the next nodes
-    workset = [node for source_node_id in source_node_ids for node in graph.get_next_nodes(source_node_id)]
```

Comment:
I would be tempted to think that there was a reason why this was here. Maybe we don't want to start from all `source_node_ids`? Is there a particular reason you deleted this? The reason why I am asking is because  regressions here cannot be caught by any test (the compiler won't fail most likely but maybe it won't optimize that much).